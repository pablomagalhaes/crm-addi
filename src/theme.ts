import { DefaultTheme } from 'styled-components';

const theme: DefaultTheme = {
  color: {
    white: '#FFF',
    black: '#424242',

    background: '#f2f2f2',
    border: 'rgba(0,0,0,0.1)',

    primary: '#6272a4',
    secondary: '#44475a',

    success: '#2AD178',
    warning: '#FABD59',
    danger: '#FF5B68',
  },

  font: {
    family: {
      primary: "'Roboto', sans-serif",
    },

    weight: {
      regular: 400,
      semiBold: 600,
    },

    color: {
      primary: '#424242',
      secondary: '#969696',
    },
  },

  borderRadius: {
    default: '3px',
  },

  breakpoint: {
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1300,
  },
};

export default theme;
