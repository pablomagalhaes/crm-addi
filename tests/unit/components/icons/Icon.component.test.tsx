import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Icon, { IconWrapper } from '@/components/icons/Icon.component';
import theme from '@/theme';

describe('<Icon />', () => {
  it('should match snapchat', () => {
    const wrapper = shallow(<Icon theme={theme} icon="id-card" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the deafult color', () => {
    const wrapper = shallow(<Icon theme={theme} icon="id-card" />);
    expect(wrapper).toHaveStyleRule('color', 'rgba(0,0,0,0.24)');
  });

  it('should have the success color when is set', () => {
    const wrapper = shallow(
      <Icon theme={theme} icon="id-card" styleType="success" />
    );
    expect(wrapper).toHaveStyleRule('color', theme.color.success);
  });

  it('should have the warning color when is set', () => {
    const wrapper = shallow(
      <Icon theme={theme} icon="id-card" styleType="warning" />
    );
    expect(wrapper).toHaveStyleRule('color', theme.color.warning);
  });

  it('should have the danger color when is set', () => {
    const wrapper = shallow(
      <Icon theme={theme} icon="id-card" styleType="danger" />
    );
    expect(wrapper).toHaveStyleRule('color', theme.color.danger);
  });
});

describe('<Paragraph />', () => {
  it('should match snapchat', () => {
    const wrapper = shallow(<IconWrapper />);
    expect(wrapper).toMatchSnapshot();
  });
});
