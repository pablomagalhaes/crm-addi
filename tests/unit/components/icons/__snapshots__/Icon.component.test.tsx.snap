// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should match snapchat 1`] = `
.c0 {
  font-size: 30px;
  color: rgba(0,0,0,0.24);
}

<FontAwesomeIcon
  border={false}
  className="c0"
  fixedWidth={false}
  flip={null}
  icon="id-card"
  inverse={false}
  listItem={false}
  mask={null}
  pull={null}
  pulse={false}
  rotation={null}
  size={null}
  spin={false}
  swapOpacity={false}
  symbol={false}
  theme={
    Object {
      "borderRadius": Object {
        "default": "3px",
      },
      "breakpoint": Object {
        "lg": 992,
        "md": 768,
        "sm": 576,
        "xl": 1300,
      },
      "color": Object {
        "background": "#ebebeb",
        "black": "#424242",
        "border": "rgba(0,0,0,0.1)",
        "danger": "#FF5B68",
        "primary": "#6272a4",
        "secondary": "#44475a",
        "success": "#2AD178",
        "warning": "#FABD59",
        "white": "#FFF",
      },
      "font": Object {
        "color": Object {
          "primary": "#424242",
          "secondary": "#969696",
        },
        "family": Object {
          "primary": "'Open Sans', sans-serif",
        },
        "weight": Object {
          "regular": 400,
          "semiBold": 600,
        },
      },
    }
  }
  title=""
  transform={null}
/>
`;

exports[`<Paragraph /> should match snapchat 1`] = `
.c0 {
  display: inline-block;
  position: relative;
}

<div
  className="c0"
/>
`;
